### PR TITLE
Add contextmenu for map-extent layercontrol

### LINKED ISSUE #926: create / enable "Zoom To..." in layer control (extent) context menu
Each extent has its own .zoomTo() method now, which should be exposed via a (sub-)layer context menu item.  

This feature gains prominence now that GeoServer is exposing multiple extents to user as a preview option.

### PR DESCRIPTION
**Pull request by AliyanH**
Closes #926

- [X] Add "Zoom to Sub Layer" + "Copy Sub Layer"
- [X] Add support for keyboard navigation
- [X] Add support for Keyboard shortcut
- [x] Add localization
- [x] Add missing localization for "Sub-Layer"
- [ ] ~Re-key copy layer/extent shortcut from 'L' to 'C'?~
- [x] when layer control context menu is open DO NOT collapse layer control?
- [x] Add tests

**Future work**
- Abiltiy to paste map-extent into layers?

### CODE REVIEW

File: src/map-extent.js [hunk 1]
```diff
@@ -71,6 +71,45 @@ export class MapExtent extends HTMLElement {
       ? getExtent(this)
       : getCalculatedExtent(this);
   }
+
+  getOuterHTML() {
+    let tempElement = this.cloneNode(true);
```
Comment (prushforth) [inline src/map-extent.js:76]: I attempted, but failed to re-write this without cloning the live element.  It's that cloning that yields the console logging due to disconnected map-input and so on (I believe).  That logging seems to interfere with the tests, and what's concerning is that it doesn't appear to happen consistently.

File: src/map-extent.js [hunk 2]
```diff
@@ -107,7 +146,7 @@ export class MapExtent extends HTMLElement {
         case 'label':
           if (oldValue !== newValue) {
             this._layerControlHTML.querySelector(
-              '.mapml-layer-item-name'
+              '.mapml-extent-item-name'
```
Comment (prushforth) [inline src/map-extent.js:149]: Good going

File: test/e2e/core/ArrowKeyNavContextMenu.test.js [hunk 3]
```diff
@@ -16,8 +16,8 @@ test.describe('Using arrow keys to navigate context menu', () => {
 
   test('Testing layer contextmenu', async () => {
     await page.waitForTimeout(500);
-    await page.click('body > mapml-viewer');
```
Comment (prushforth) [inline test/e2e/core/ArrowKeyNavContextMenu.test.js:19]: this is [not recommended by playwright](https://playwright.dev/docs/next/api/class-page#page-click), so I changed a few of them where they seemed to cause some flakiness

**Review by prushforth — APPROVED** [review]
Ready to merge, integrate into release 0.13.1

**Comment by AliyanH** [comment]
Thanks @prushforth! Will merge after finalizing https://github.com/Maps4HTML/mapml-extension/pull/69

**Comment by prushforth** [comment]
> Thanks @prushforth! Will merge after finalizing [Maps4HTML/mapml-extension#69](https://github.com/Maps4HTML/mapml-extension/pull/69)

We should merge this when it's ready so that we can release it as 0.13.1 ([with a small other GeoServer-related change that wouldn't pass CI reliably without these commits](https://github.com/Maps4HTML/Web-Map-Custom-Element/pull/932)) with GeoServer 2.25.0, [which will happen March 18th](https://github.com/geosolutions-it/geoserver/pull/363#issuecomment-1986174593).  I could merge that change without this, of course, but it was easier to rebase it on the work done in this branch which eliminates some test flakiness (there's always a new source of flakiness, for some reason).

The GeoServer viewer should not depend on the extension, but I agree the extension needs this update.

I will open a GeoServer PR this week to include the new viewer, which we can update until Friday.

**Comment by prushforth** [comment]
I'll cut a 0.13.1 release later today, hopefully in time to make the GeoServer 2.25.0 release cut-off, tbd.

**Pull request merged**